src/hotspot/share/opto/memnode.cpp

```diff
@@ -2790,22 +2790,24 @@ class MergePrimitiveStores : public StackObj {
 private:
   PhaseGVN* const _phase;
   StoreNode* const _store;
+  enum DataOrder { Unknown, Forward, Reverse};
+  DataOrder  _value_order;
```

Comment:
```suggestion
  // State machine with initial state Unknown
  // Allowed transitions:
  //   Unknown  -> Forward
  //   Unknown  -> Backward
  //   Forward  -> Forward
  //   Backward -> Backward
  enum DataOrder { Unknown, Forward, Reverse };
  DataOrder  _value_order;
```

Also: call it either consistently data-order or value-order ;)

Reply:
Thanks for comment. I rename the enum to ValueOrder, and rename Reverse to Backward.